components/motor/fake/motor.go

```diff
@@ -107,20 +119,31 @@ func (m *Motor) Reconfigure(ctx context.Context, deps resource.Dependencies, con
 	}
 	if newConf.BoardName != "" {
 		m.Board = newConf.BoardName
-		b, err := board.FromDependencies(deps, m.Board)
+		b, err = board.FromDependencies(deps, m.Board)
 		if err != nil {
 			return err
 		}
-		if newConf.Pins.PWM != "" {
-			m.PWM, err = b.GPIOPinByName(newConf.Pins.PWM)
-			if err != nil {
-				return err
-			}
-			if err = m.PWM.SetPWMFreq(ctx, newConf.PWMFreq, nil); err != nil {
-				return err
-			}
+	} else {
+		m.Logger.Info("board not provided, using a fake board")
+		m.Board = "fakeboard"
+		b, err = fakeboard.NewBoard(ctx, fakeBoardConf, m.Logger)
+		if err != nil {
+			return err
 		}
 	}
+
+	pwmPin := "1"
+	if newConf.Pins.PWM != "" {
+		pwmPin = newConf.Pins.PWM
+	}
+	m.PWM, err = b.GPIOPinByName(pwmPin)
+	if err != nil {
+		return err
+	}
+	if err = m.PWM.SetPWMFreq(ctx, newConf.PWMFreq, nil); err != nil {
```

Comment:
`m.PWM` is initialized on line 139, used on line 143, and never used again. Why is it a field within `m` at all, rather than a local variable?

...for that matter, why does a fake motor need to set a real PWM pin anyway? I wonder if this entire paragraph can be removed. and if so, perhaps the previous paragraph can be removed, too, which gets rid of the trouble with a global `Board` variable.

Reply:
it's used in a couple of tests. as is the other seemingly dead code. I think I'm going to leave it for now as that's not the goal of this ticket

Reply:
these seemingly dead code snippets are helpful for testing, and making sure that your board's pin functions are not causing a panic before you hook up the real motor. It's a weird testing case, most of this is dead code, but I'd like anything that tests out pin config setting to stay for us devs. We can remove this in the far future. 

The use case that is really interesting with fake motor is using maxrpm and ticksperotation to simulate movement. Ultimately, that's what I want it to look like, but incremental changes are okay.

I don't think we're calling the PWM anywhere though, to have a real board send the fake motor a PWM command that someone could scope. So it can stay local unless I'm mistaken and that code des exist somewhere.

Also - if any tests are casting a fake board and checking for a PWM pin reconfiguration, that will break. Up to you i you want to tackle that Martha!

Reply:
`TestRobotReconfigure` in robot/impl/robot_reconfigure_test.go tests the pwmFreq. I could try and restructure that test to not use fake motor if anyone feels strongly about it, but it seems like fake motor can be as similar as possible to a real motor to simulate as many cases as possible. so I don't see a huge issue with leaving as is but let me know

Reply:
Leave it as is - fakes are there for testing especially with the robot code, and so this model should be in service of that.